core/templates/dev/head/pages/exploration-editor-page/feedback-tab/services/thread-data.service.ts

```diff
@@ -27,162 +27,158 @@ require(
-            suggestionsResponse.suggestions.length) {
-          $log.error('Number of suggestion threads doesn\'t match number of' +
-                     'suggestion objects');
-          return _data;
-        }
-        // TODO(brianrodri@): Move this pairing logic into the backend.
-        for (let threadDict of threadsResponse.suggestion_thread_dicts) {
-          for (let suggestionDict of suggestionsResponse.suggestions) {
-            if (threadDict.thread_id === suggestionDict.suggestion_id) {
-              _data.suggestionThreads.push(
-                SuggestionThreadObjectFactory.createFromBackendDicts(
-                  threadDict, suggestionDict));
-              break;
-            }
-          }
-        }
-        return _data;
-      });
+    let setSuggestionThreadFromBackendDicts = function(
+        threadBackendDict, suggestionBackendDict) {
+      let threadId = threadBackendDict.thread_id;
+      return threadsById[threadId] =
+        SuggestionThreadObjectFactory.createFromBackendDicts(
+          threadBackendDict, suggestionBackendDict);
     };
 
-    var _fetchMessages = function(threadId) {
-      return $http.get(_THREAD_HANDLER_PREFIX + threadId).then(function(res) {
-        getThreadById(threadId).setMessages(res.data.messages);
-      });
+    let getThreadIdFromSuggestionBackendDict = function(suggestionBackendDict) {
+      return suggestionBackendDict.suggestion_id;
     };
 
     return {
+      getThread: function(threadId) {
+        return getThreadById(threadId);
+      },
+
       fetchThreads: function() {
-        return _fetchThreads();
+        let suggestionPromise = $http.get(SUGGESTION_LIST_HANDLER_URL, {
+          params: {target_type: 'exploration', target_id: expId}
+        });
+        let threadPromise = $http.get(THREAD_LIST_HANDLER_URL);
+
+        return $q.all([suggestionPromise, threadPromise]).then(response => {
+          let [suggestionResponse, threadResponse] = response.map(r => r.data);
+
+          let suggestionBackendDictsByThreadId = {};
+          suggestionResponse.suggestions.forEach(backendDict => {
+            let threadId = getThreadIdFromSuggestionBackendDict(backendDict);
+            suggestionBackendDictsByThreadId[threadId] = backendDict;
+          });
+
+          return {
+            feedbackThreads: threadResponse.feedback_thread_dicts.map(
+              setFeedbackThreadFromBackendDict),
+            suggestionThreads: threadResponse.suggestion_thread_dicts.map(
+              threadBackendDict => setSuggestionThreadFromBackendDicts(
+                threadBackendDict,
+                suggestionBackendDictsByThreadId[threadBackendDict.thread_id]))
+          };
+        });
       },
-      fetchMessages: function(threadId) {
-        return _fetchMessages(threadId);
+
+      fetchMessages: function(thread) {
+        let threadId = thread.threadId;
+
+        return $http.get(THREAD_HANDLER_URL + '/' + threadId).then(response => {
```

Comment:
Use UrlInterpolationService to construct the URL.

Reply:
Done.